File: 1363-greatest-english-letter-in-upper-and-lower-case/greatest-english-letter-in-upper-and-lower-case.java
class Solution {
    public static String greatestLetter(String s) {
        Set<Character> lower = new HashSet<>();
        Set<Character> upper = new HashSet<>();
        
        for (char c : s.toCharArray()) {
            if (Character.isLowerCase(c)) lower.add(c);
            else upper.add(c);
        }
        
        char result = 0;
        for (char c = 'A'; c <= 'Z'; c++) {
            if (upper.contains(c) && lower.contains(Character.toLowerCase(c))) {
                result = c;
            }
        }
        
        return result == 0 ? "" : String.valueOf(result);
    }
}

File: 1363-greatest-english-letter-in-upper-and-lower-case/greatest-english-letter-in-upper-and-lower-case.cpp
class Solution {
public:
    string greatestLetter(string s) {
    unordered_set<char> lower, upper;
    
    for (char c : s) {
        if (islower(c)) lower.insert(c);
        else upper.insert(c);
    }
    
    char result = '\0';
    
    for (char c = 'A'; c <= 'Z'; ++c) {
        if (upper.count(c) && lower.count(c + 32)) { 
            result = c; 
        }
    }
    
    return result == '\0' ? "" : string(1, result);
}
};